src/perl/lib/wtsi_clarity/epp/generic/worksheet_attacher.pm

```diff
@@ -224,14 +225,26 @@ sub _get_TECAN_file_content_per_URI {
       my $input_buffer_string  = qq{A;BUFF;;96-TROUGH;$inp_loc_dec;;$buffer_volume};
       my $output_buffer_string = qq{D;$output_barcode;;$output_type;$out_loc_dec;;$buffer_volume};
 
-      push @sample_output, $input_sample_string;
-      push @sample_output, $output_sample_string;
-      push @sample_output, $w_string;
-      push @buffer_output, $input_buffer_string;
-      push @buffer_output, $output_buffer_string;
-      push @buffer_output, $w_string;
+      push @sample_output, {
+        output_location => $out_loc_dec,
+        output          => join "\n", $input_sample_string, $output_sample_string, $w_string,
+      };
+
+      push @buffer_output, {
+        output_location => $out_loc_dec,
+        output => join "\n", $input_buffer_string, $output_buffer_string, $w_string,
+      };
     }
   }
+
+  @sample_output = map { $_->{'output'} } sort {
+    $a->{'output_location'} <=> $b->{'output_location'}
+  } @sample_output;
```

Comment:
The code suggests the 'output_location' should be a number, because it compares the 2 values with the '<=>' operator, but there is no check for this in the code. I would add this validation right after getting the value.